### ISSUE TITLE
Files missing in firebase@9.15.0

### ISSUE THREAD

**Issue by jwalton** [firebase/firebase-js-sdk#6887]
### Describe your environment

  * Operating System version: MacOS 13.0
  * Browser version: Any
  * Firebase SDK version: 9.15.0
  * Firebase Product: auth, possibly others

#### Steps to reproduce:

```sh
$ npm install firebase@9.14.0
$ ls node_modules/firebase/auth/dist
analytics        auth             firestore        index.cjs.js.map index.mjs        messaging        storage
app              compat           functions        index.esm.js     index.mjs.map    performance
app-check        database         index.cjs.js     index.esm.js.map installations    remote-config

$ npm install firebase@9.15.0
$ ls node_modules/firebase/auth/dist
analytics        auth             esm              index.cjs.js     index.mjs.map    performance
app              compat           firestore        index.cjs.js.map installations    remote-config
app-check        database         functions        index.mjs        messaging        storage
```

Note that `index.esm.js` is missing, however this file is still referenced in node_modules/firebase/auth/package.json:

```sh
$ cat node_modules/firebase/auth/package.json
{
  "name": "firebase/auth",
  "main": "dist/index.cjs.js",
  "browser": "dist/esm/index.esm.js",
  "module": "dist/esm/index.esm.js",
  "typings": "dist/auth/index.d.ts"
}
```

As a result, if you for example `import { getAuth } from 'firebase/auth';`  and you use an ESM-based bundler like vite, then the build will work in 9.14.0, but fail in 9.15.0.

**Comment by hsubox76** [comment]
The path is in package.json is `dist/esm/index.esm.js`, note the `esm` subdirectory, which I do see in your `ls` results. Does the file exist in that subdirectory?

If so, and the build is still failing, can you share the error message and a minimal repro of the project it won't build in?

**Comment by hsubox76** [comment]
Note to self, possibly caused by https://github.com/firebase/firebase-js-sdk/pull/6826

**Comment by jwalton** [comment]
My humble appologies.  I'm afraid I'm unable to reproduce this.  I'm 99% sure that what happened here is, I had left a `vite serve` open in a shell, I upgraded to the latest firebase, and the vite started complaining that node_modules/firebase/auth/dist/index.esm.js couldn't be found.  I had just merged a lockfile maintenance PR from renovate, so I tried downgrading firebase and of course this fixed the problem.  But the real problem was that I needed to restart my vite server.

So sorry for jumping the gun and opening an issue.  :(
